### PR TITLE
minor modifications in HypercubicGird.hpp and Lattice.hpp

### PR DESCRIPTION
**Pull request by galvanixer**
1. GetJump(i,j) moved from Lattice to HypercubicGrid
2. GetJump() updated to handle for open boundary conditions and odd lattice size.
3. Testing for GetJump() added to unit tests.
4. Minor change in GetMomenta.
5. clang-format

### CODE REVIEW

File: include/bwsl/Lattice.hpp [hunk 1]
```diff
@@ -139,18 +135,27 @@ class Lattice : public HyperCubicGrid
...
+  /// NOTE: the positions stored are in real space.
   [[nodiscard]] auto ComputePositions(Bravais const& bravais) const
     -> std::vector<realvec_t>;
 
-  /// Compute the distance vectors
+  /// Compute the distance vectors.
+  /// It is composed of vectors in real space between site 0 and site i.
+  /// Here, i ∈ [0,1,...,N] where N is the largest site index.
+  /// For closed boundary conditions each vector is chosen in such a manner
+  /// that it represents the shortest distance between the two sites.
```
Comment (gdmsl) [inline include/bwsl/Lattice.hpp]:
```suggestion
  /// that it represents the shortest distance between the first site and all
  /// the periodic images of the second one (or vice versa).
  /// [minimum distance convention]
```

File: include/bwsl/Lattice.hpp [hunk 2]
```diff
@@ -139,18 +135,27 @@ class Lattice : public HyperCubicGrid
...
   [[nodiscard]] auto ComputeVectors(Bravais const& bravais) const
     -> std::vector<realvec_t>;
 
   /// Compute the vector of distances
```
Comment (gdmsl) [inline include/bwsl/Lattice.hpp]:
```suggestion
  /// Compute the vector of distances respecting the minimum
  /// distance convention (if with closed boundaries).
```

File: include/bwsl/HyperCubicGrid.hpp [hunk 3]
```diff
@@ -237,6 +241,31 @@ HyperCubicGrid::GetIndividualIndices(size_t pair) const
...
+{
+  assert(IndexIsValid(a) && IndexIsValid(b));
+  auto cb = GetCoordinates(b);
+  subtract_into(cb, GetCoordinates(a));
+
+  if (HasClosedBoundaries()) {
+    for (auto i = 0UL; i < GetDim(); i++) {
+      auto& cbi = cb[i];
+      auto si = GetSize()[i];
+
```
Comment (gdmsl) [inline include/bwsl/HyperCubicGrid.hpp]:
```suggestion
      // half distance (integer division)
      auto si_half = static_cast<int>(si / 2);
```

File: include/bwsl/HyperCubicGrid.hpp [hunk 4]
```diff
@@ -237,6 +241,31 @@ HyperCubicGrid::GetIndividualIndices(size_t pair) const
...
+      if (cbi > static_cast<int>(si / 2)) {
```
Comment (gdmsl) [inline include/bwsl/HyperCubicGrid.hpp]:
```suggestion
      if (cbi > si_half) {
```

File: include/bwsl/HyperCubicGrid.hpp [hunk 5]
```diff
@@ -237,6 +241,31 @@ HyperCubicGrid::GetIndividualIndices(size_t pair) const
...
+        cbi -= si;
+      }
+      if ((cbi < -static_cast<int>(si / 2)) ||
```
Comment (gdmsl) [inline include/bwsl/HyperCubicGrid.hpp]:
```suggestion
      if ((cbi < -si_half) ||
```

File: include/bwsl/HyperCubicGrid.hpp [hunk 6]
```diff
@@ -237,6 +241,31 @@ HyperCubicGrid::GetIndividualIndices(size_t pair) const
...
+          (si % 2 == 0 && cbi == -static_cast<int>(si / 2))) {
```
Comment (gdmsl) [inline include/bwsl/HyperCubicGrid.hpp]:
```suggestion
          (si % 2 == 0 && cbi == -si_half)) {
```

**Review by gdmsl — CHANGES_REQUESTED** [review]
Everything looks OK. Just added few stylistic changes around.

**Review by gdmsl — APPROVED** [review]
